content/concepts/consensus-network/consensus-protections.md

```diff
@@ -0,0 +1,67 @@
+
+This page describes some of the types of challenges that the XRP Ledger Consensus Protocol faces and how it handles them.
+
+## Individual Validators Misbehaving
+
+_Validators_ are servers who actively contribute to the process of deciding each new ledger version. Validators only have an influence over servers configured to trust them (including indirectly). Consensus can continue even if some validators are misbehaving, including a large variety of failure cases, such as:
+
+- Being unavailable or overloaded.
+- Being partially disconnected from the network, so their messages reach only a subset of participants without delay.
+- Intentionally behaving with intent to defraud others or halt the network.
+- Behaving maliciously as a result of pressure from outside factors, such as threats from an oppressive government.
+- Accidentally sending confusing or malformed messages due to a bug or outdated software.
+
+In general, consensus can continue without problems as long as only a small percentage (less than about 20%) of trusted validators are misbehaving at a given time. (For the exact percentages and the math behind them, see the latest [Consensus Research](consensus-research.html).)
+
+If more than about 20% of validators are unreachable or not behaving properly, the network fails to reach a consensus. During this time, new transactions can be tentatively processed, but new ledger versions cannot be validated, so those transactions' final outcomes are not certain. In this situation, it would become immediately obvious that the XRP Ledger is unhealthy, prompting intervention from human participants who can decide whether to wait, or reconfigure their set of trusted validators.
+
+The only way to confirm invalid transactions—such as those spending money that has already been spent, or otherwise breaking the rules of the network—would be if more than about 80% of trusted validators colluded to confirm those transactions. In other words, a large majority of trusted validators would have to misbehave in the exact same way. With dozens of trusted validators run by different people and businesses in different parts of the world, this would be very difficult to achieve intentionally.
+
+
+## Software Vulnerabilities
+
+As with any software system, bugs (or intentionally malicious code) in the implementation of the XRP Ledger Consensus Protocol, commonly deployed software packages, or their dependencies, are a problem to be taken seriously. Even bugs that simply cause a server to crash when it sees carefully-crafted inputs can be abused to disrupt the progress of the network. Ripple has a number of precautions in place to address this threat, including:
```

Comment:
Consider: 

```
< carefully-crafted inputs
---
> carefully crafted inputs

```

CMOS 7.86: _Compounds formed by an adverb ending in ly plus an adjective or participle (such as largely irrelevant or smartly dressed) are not hyphenated ..._